udmis/src/main/java/com/google/bos/udmi/service/core/ProvisioningEngine.java

```diff
@@ -112,6 +113,42 @@ private boolean shouldProvision(Date generation, CloudModel cloudModel) {
     return TRUE_OPTION.equals(ifNotNullGet(cloudModel.metadata, m -> m.get(UDMI_PROVISION_ENABLE)));
   }
 
+  private void gatewayDiscovery(DiscoveryEvents discoveryEvent, CloudModel gatewayModel,
+      Envelope envelope, String registryId, String deviceId, Date generation, String expectedId) {
+    Set<String> deviceIds = refreshModelDevices(registryId, deviceId, generation, gatewayModel);
+    if (deviceIds == null) {
```

Comment:
doesn't `shouldProvision` work here too?  Seems like that should be constant in both cases.